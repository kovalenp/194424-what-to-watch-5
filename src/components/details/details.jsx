import React from "react";

import { movieProps } from "../../validation/propTypes";

const Details = (props) => {

  const { movie } = props;

  return (
    <>
      <div className="movie-card__text movie-card__row">
        <div className="movie-card__text-col">
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Director</strong>
            <span className="movie-card__details-value">{movie.director}</span>
          </p>
          <div className="movie-card__details-item">
            <strong className="movie-card__details-name">Starring</strong>
            <ul className="movie-card__details-value" style={{ padding: 0 }}>
              {movie.cast.map((actor, i) => (
                <li style={{ listStyleType: `none` }} key={i}>{actor}</li>
              ))}
            </ul>
          </div>
        </div>

        <div className="movie-card__text-col">
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Run Time</strong>
            <span className="movie-card__details-value">1h 39m</span>
          </p>
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Genre</strong>
            <span className="movie-card__details-value">{movie.genre}</span>
          </p>
          <p className="movie-card__details-item">
            <strong className="movie-card__details-name">Released</strong>
            <span className="movie-card__details-value">{movie.year}</span>
          </p>
        </div>
      </div>
    </>
  );
};

Details.propTypes = {
  movie: movieProps
};

export default Details;
