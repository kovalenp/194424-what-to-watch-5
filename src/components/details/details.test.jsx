import React from "react";
import renderer from "react-test-renderer";
import {Details} from "./details";
import movie from "../../../test/mock/movie";

it(`Details test to meet B21 criteria`, () => {
  const tree = renderer
    .create(
        <Details
          movie={movie}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
