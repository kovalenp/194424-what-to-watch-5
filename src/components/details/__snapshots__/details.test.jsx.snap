// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details test to meet B21 criteria 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Nicolas Winding Refn
      </span>
    </p>
    <div
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <ul
        className="movie-card__details-value"
        style={
          Object {
            "padding": 0,
          }
        }
      >
        <li
          style={
            Object {
              "listStyleType": "none",
            }
          }
        >
          Tom Hardy
        </li>
        <li
          style={
            Object {
              "listStyleType": "none",
            }
          }
        >
          Kelly Adams
        </li>
        <li
          style={
            Object {
              "listStyleType": "none",
            }
          }
        >
          Luing Andrews
        </li>
      </ul>
    </div>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        1h 39m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Action
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      />
    </p>
  </div>
</div>
`;
